src/App.tsx

```diff
@@ -2,14 +1,80 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
+
+function filterPerson(arr: Person[], query: string) {
+  return query ? [...arr].filter(elem => elem.name.includes(query)) : [...arr];
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [title, setTitle] = useState('No selected person');
+
+  const [showArray, setShowArray] = useState(false);
+  const showMessageRef = useRef(false);
+
+  const updatedPersons = useMemo(() => {
+    return filterPerson(peopleFromServer, appliedQuery);
+  }, [appliedQuery]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const handleChangeInputName = (
```

Comment:
Review and adjust method names to align with recommended naming conventions. For example, `handleChangeInputName` could be simplified to `handleInputChange`.